### PR TITLE
Shadowbeam

### PR DESCRIPTION
**Pull request by phantamanta44**
Adds the Rod of the Corrupted Shadow, plus corresponding entities, sounds, recipes, etc.
It's effectively the shadowbeam focus from TTKAMI, except it uses mana and isn't an end-tier item.

I have no idea why the other commit is there again under a different hash, but I'll just roll with it...?
Something probably went wack during rebase...

### CODE REVIEW

File: src/main/java/vazkii/botania/common/entity/EntityShadowBeam.java [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+import net.minecraft.util.DamageSource;
+import net.minecraft.util.MathHelper;
+import net.minecraft.util.MovingObjectPosition;
+import net.minecraft.util.Vec3;
+import net.minecraft.world.World;
+import net.minecraftforge.common.util.ForgeDirection;
+import vazkii.botania.common.Botania;
+import vazkii.botania.common.core.helper.Vector3;
+
+public class EntityShadowBeam extends EntityThrowable {
```
Comment (Vazkii) [inline src/main/java/vazkii/botania/common/entity/EntityShadowBeam.java]: Use EntityThrowableFake instead.

File: src/main/java/vazkii/botania/common/item/lens/ItemLens.java [hunk 2]
```diff
@@ -319,6 +319,6 @@ public ItemStack setCompositeLens(ItemStack sourceLens, ItemStack compositeLens)
 
 	@Override
 	public boolean shouldPull(ItemStack stack) {
-		return stack.getItemDamage() != STORM;
+		return stack.getItemDamage() < 5000;
```
Comment (Vazkii) [inline src/main/java/vazkii/botania/common/item/lens/ItemLens.java]: ?

File: src/main/resources/assets/botania/lang/en_US.lang [hunk 3]
```diff
@@ -1433,6 +1434,7 @@ item.botania:slimeBottle.name=Slime in a Bottle
 item.botania:starSword.name=Starcaller
 item.botania:exchangeRod.name=Rod of the Shifting Crust
 item.botania:magnetRingGreater.name=Greater Ring of Magnetization
+item.botania:shadowRod.name=Rod of the Corrupted Shadow
```
Comment (Vazkii) [inline src/main/resources/assets/botania/lang/en_US.lang]: Names of rods must resemble some sort of possible geological location or event, this name doesn't fit.

File: src/main/java/vazkii/botania/common/crafting/ModCraftingRecipes.java [hunk 4]
```diff
@@ -1899,6 +1900,15 @@ else addOreDictRecipe(new ItemStack(ModItems.cacophonium),
 				'S', "stone",
 				'R', LibOreDict.RUNE[12]);
 		recipeExchangeRod = BotaniaAPI.getLatestAddedRecipe();
+		
+		// Rod of the Corrupted Shadow Recipe
+		addOreDictRecipe(new ItemStack(ModItems.shadowRod),
+				" DI", " TD", "G  ",
+				'T', LibOreDict.DREAMWOOD_TWIG,
+				'D', LibOreDict.DRAGONSTONE,
+				'I', LibOreDict.GAIA_INGOT,
```
Comment (Vazkii) [inline src/main/java/vazkii/botania/common/crafting/ModCraftingRecipes.java]: No. The ingot should not have any recipes.

File: src/main/resources/assets/botania/lang/en_US.lang [hunk 5]
```diff
@@ -2479,6 +2481,13 @@ botania.page.exchangeRod1=Right clicking on a block in the world will start a la
...
 botania.page.exchangeRod3=Bootlegged equal trade
 
+# -- ROD OF THE CORRUPTED SHADOW
+botania.entry.shadowRod=Rod of the Corrupted Shadow
+botania.page.shadowRod0=Sometimes, fighting off monsters with conventional tools just seems too dull. Even the most powerful of elven mages need to have their fun, right? To this point, the &1Rod of the Corrupted Shadow&0 was created.<br>By combining a powerful source of magic, such as a &1Dragonstone&0, with a material that absorbs most energies, namely the &1Gaia Spirit Ingot&0,
+botania.page.shadowRod1=a large excess of neutral magical energy can be produced, which the rod then focuses into a beam you can direct towards unfriendly creatures.<br>The beam emitted by the rod will travel in a straight line for a good distance until it collides with a solid surface, at which point it will bounce similarly to a ray of light striking a mirror.
```
Comment (Vazkii) [inline src/main/resources/assets/botania/lang/en_US.lang]: Text shouldn't go from one page to another like that.

File: src/main/resources/assets/botania/lang/en_US.lang [hunk 6]
```diff
@@ -2479,6 +2481,13 @@ botania.page.exchangeRod1=Right clicking on a block in the world will start a la
...
+botania.page.shadowRod2=Naturally, such power doesn't come without a hefty price. The rod will consume a good bit of &4mana&0 as it's firing, regardless of whether it hits anything or not.
+botania.page.shadowRod3=KAMIhameha!
```
Comment (Vazkii) [inline src/main/resources/assets/botania/lang/en_US.lang]: No.

File: src/main/java/vazkii/botania/common/item/lens/ItemLens.java (hunk 2 above)
Comment (phantamanta44) [inline src/main/java/vazkii/botania/common/item/lens/ItemLens.java]: I was initially going to use a beam of mana bursts, but then I found that bad things happened if you directed it towards a mana acceptor.

**Comment by Vazkii** [comment]
Your sound crashed winamp, you probably want to look into that.

**Comment by phantamanta44** [comment]
Well then. I have a lot to do tomorrow :v

File: private.properties.example [hunk 7]
```diff
@@ -1,4 +1,5 @@
 host=
 user=
 pass=
-cfkey=
\ No newline at end of file
+cfkey=
+dir_forgecraft=../Build Output/Botania/
```
Comment (Vazkii) [inline private.properties.example]: Rebase and remove these old changes please.

File: src/main/java/vazkii/botania/common/item/rod/ItemShadowRod.java [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+	@Override
+	public int getMaxItemUseDuration(ItemStack par1ItemStack) {
+		return 72000;
+	}
+
+	@Override
+	public void onUsingTick(ItemStack stack, EntityPlayer player, int count) {
+		if(!player.worldObj.isRemote && ManaItemHandler.requestManaExact(stack, player, COST, true)) {
+			if (count % 3 == 0)
```
Comment (Vazkii) [inline src/main/java/vazkii/botania/common/item/rod/ItemShadowRod.java]: Fix the parenthesis style, botania doesn't have a space after the if/for/while.

File: src/main/java/vazkii/botania/common/entity/EntityShadowBeam.java [hunk 9]
```diff
@@ -0,0 +1,308 @@
+/**
+ * This class was created by <Phanta>. It's distributed as
+ * part of the Botania Mod. Get the Source Code in github:
+ * https://github.com/Vazkii/Botania
+ * 
+ * Botania is Open Source and distributed under the
+ * Botania License: http://botaniamod.net/license.php
+ * 
+ * File Created @ [Aug 24, 2015, 12:31:50 AM (CDT)]
```
Comment (Vazkii) [inline src/main/java/vazkii/botania/common/entity/EntityShadowBeam.java]: Convert this to GMT for consistency.

File: src/main/resources/assets/botania/lang/en_US.lang (hunk 3 above)
Comment (TheWhiteWolves) [inline src/main/resources/assets/botania/lang/en_US.lang]: how about "Rod of Crystalline Refraction"

**Comment by phantamanta44** [comment]
Wait, rebase just re-did all your commits under different hashes?
Something's wrong here.

**Comment by phantamanta44** [comment]
Fixing things on another branch.

File: src/main/java/vazkii/botania/common/entity/EntityShadowBeam.java (hunk 1 above)
Comment (phantamanta44) [inline src/main/java/vazkii/botania/common/entity/EntityShadowBeam.java]: What exactly is "EntityThrowableFake"?

**Pull request closed without merging**